### PR TITLE
feat: Package matplotlib and ui JS with wheel

### LINKED ISSUE #194: Package JS with python for plugins
Currently only plotly-express packages the JS with the plugin, and registers the JS from python. Switch matplotlib, deephaven.ui over to do that as well.

We should be able to also try and register, and if the JS plugin is already registered then just warn instead of throwing.

Related to #139

### LINKED ISSUE #316: Remove DEEPHAVEN_ENABLE_PY_JS
https://github.com/deephaven/deephaven-plugins/pull/285 provides a bad experience for community developers and users. Whenever possible, we should be leaning into community being the "default" configuration, with specific hooks that satisfy use cases that DHE may need. In this specific case, I think the impetus should be on DHE to opt-out instead of making community users opt-in.

From the context of python, it may make sense to check for an enterprise-only python package, or file on the filesystem, to determine that this is in enterprise. A DHE-specific environment variable might be acceptable too, but that would be less sticky than an enterprise package or file.

### PR DESCRIPTION
**Pull request by jnumainville**
Fixes #194, fixes #316 

Packages matplotlib and ui JS with wheel using new util packages. I also took the opportunity to refactor those two a bit to match the plotly express package by moving registration out of the `__init__.py` files. Also refactored plotly express.

I've verified the installed wheels work properly with basic examples.

By the way @devinrsmith I invited deephaven as an owner of the newly uploaded packages 
https://pypi.org/project/deephaven-plugin-utilities/
https://pypi.org/project/deephaven-plugin-packaging/

### CODE REVIEW

File: plugins/plotly-express/setup.py [hunk 1]
```diff
@@ -1,38 +1,10 @@
...
-    )
-    # it is assumed that there is only one tarball in the directory
-    files = os.listdir(build_dir)
-    for file in files:
-        subprocess.run(["tar", "-xzf", file], cwd=build_dir, check=True)
-        os.remove(os.path.join(build_dir, file))
 
-    # move the package directory to the expected package location
-    shutil.move(package_dir, dest_dir)
+package_js(js_dir, dest_dir)
```
Comment (mofojed) [inline plugins/plotly-express/setup.py:8]: Nice.

File: plugins/ui/src/deephaven/ui/__init__.py [hunk 2]
```diff
@@ -4,17 +4,9 @@
 The API is designed to be similar to React, but with some differences to make it more Pythonic.
 """
 
-from deephaven.plugin import Registration, Callback
 from .components import *
 from .elements import *
 from .hooks import *
 from .object_types import *
 
 __version__ = "0.8.0.dev0"
```
Comment (mofojed) [inline plugins/ui/src/deephaven/ui/__init__.py:12]: So I know this is a rejected standard: https://peps.python.org/pep-0396/
But at the same time - what _is_ the standard now? Closest thing I can find is this: https://packaging.python.org/en/latest/guides/single-sourcing-package-version/
Which lists 7 (!) different ways to do it, many of them which actually use `__version__` :laughing:  

I'd be good just putting the version in the setup.cfg, or whatever, we should just be consistent. Doesn't have to be a change in this PR.

**Review by mofojed — APPROVED** [review]

**Review by mofojed — CHANGES_REQUESTED** [review]
You'll need to update the Dockerfile configs now that you're registering the plugin with python.

Try running `npm run docker` to verify it works/starts up correctly. Right now it's throwing this error:

```
deephaven-plugins-deephaven-plugins-1  | Initiating shutdown due to: Uncaught exception in thread main
deephaven-plugins-deephaven-plugins-1  | java.lang.RuntimeException: Error in Python interpreter:
deephaven-plugins-deephaven-plugins-1  | Type: <class 'RuntimeError'>
deephaven-plugins-deephaven-plugins-1  | Value: Failed to register @deephaven/js-plugin-ui@0.8.0 embedded in Python plugin: java.lang.IllegalArgumentException: js plugin with name '@deephaven/js-plugin-ui' already exists. See https://github.com/deephaven/deephaven-core/issues/3048
deephaven-plugins-deephaven-plugins-1  |        at io.deephaven.server.jetty.JsPluginsZipFilesystem.checkExisting(JsPluginsZipFilesystem.java:82)
deephaven-plugins-deephaven-plugins-1  |        at io.deephaven.server.jetty.JsPluginsZipFilesystem.add(JsPluginsZipFilesystem.java:61)
deephaven-plugins-deephaven-plugins-1  |        at io.deephaven.server.jetty.JsPlugins.register(JsPlugins.java:38)
deephaven-plugins-deephaven-plugins-1  |        at io.deephaven.server.plugin.PluginRegistrationVisitor.visit(PluginRegistrationVisitor.java:46)
deephaven-plugins-deephaven-plugins-1  |        at io.deephaven.server.plugin.PluginRegistrationVisitor.visit(PluginRegistrationVisitor.java:19)
deephaven-plugins-deephaven-plugins-1  |        at io.deephaven.plugin.js.JsPlugin.walk(JsPlugin.java:105)
deephaven-plugins-deephaven-plugins-1  |        at io.deephaven.server.plugin.PluginRegistrationVisitor.register(PluginRegistrationVisitor.java:35)
deephaven-plugins-deephaven-plugins-1  |        at io.deephaven.server.plugin.PluginRegistration$Counting.visit(PluginRegistration.java:70)
deephaven-plugins-deephaven-plugins-1  |        at io.deephaven.server.plugin.PluginRegistration$Counting.visit(PluginRegistration.java:46)
deephaven-plugins-deephaven-plugins-1  |        at io.deephaven.plugin.js.JsPlugin.walk(JsPlugin.java:105)
deephaven-plugins-deephaven-plugins-1  |        at io.deephaven.server.plugin.PluginRegistration$Counting.register(PluginRegistration.java:53)
deephaven-plugins-deephaven-plugins-1  |        at io.deephaven.server.plugin.python.CallbackAdapter.registerJsPlugin(CallbackAdapter.java:25)
deephaven-plugins-deephaven-plugins-1  |        at org.jpy.PyLib.callAndReturnValue(Native Method)
deephaven-plugins-deephaven-plugins-1  |        at org.jpy.PyProxyHandler.invoke(PyProxyHandler.java:120)
deephaven-plugins-deephaven-plugins-1  |        at io.deephaven.server.plugin.python.$Proxy16.initialize_all_and_register_into(Unknown Source)
deephaven-plugins-deephaven-plugins-1  |        at io.deephaven.server.plugin.python.PythonPluginsRegistration.registerInto(PythonPluginsRegistration.java:33)
deephaven-plugins-deephaven-plugins-1  |        at io.deephaven.server.plugin.PluginRegistration.registerAll(PluginRegistration.java:41)
deephaven-plugins-deephaven-plugins-1  |        at io.deephaven.server.runner.DeephavenApiServer.run(DeephavenApiServer.java:150)
deephaven-plugins-deephaven-plugins-1  |        at io.deephaven.server.jetty.JettyMain.main(JettyMain.java:24)
deephaven-plugins-deephaven-plugins-1  | 
deephaven-plugins-deephaven-plugins-1  | Line: 57
deephaven-plugins-deephaven-plugins-1  | Namespace: _register_js
deephaven-plugins-deephaven-plugins-1  | File: /opt/deephaven/venv/lib/python3.10/site-packages/deephaven/plugin/utilities/dhe_safe_callback_wrapper.py
deephaven-plugins-deephaven-plugins-1  | Traceback (most recent call last):
deephaven-plugins-deephaven-plugins-1  |   File "/opt/deephaven/venv/lib/python3.10/site-packages/deephaven_internal/plugin/__init__.py", line 18, in initialize_all_and_register_into
deephaven-plugins-deephaven-plugins-1  |   File "/opt/deephaven/venv/lib/python3.10/site-packages/deephaven_internal/plugin/register.py", line 20, in initialize_all_and_register_into
deephaven-plugins-deephaven-plugins-1  |   File "/opt/deephaven/venv/lib/python3.10/site-packages/deephaven/plugin/__init__.py", line 95, in register_all_into
deephaven-plugins-deephaven-plugins-1  |   File "/opt/deephaven/venv/lib/python3.10/site-packages/deephaven/ui/_register.py", line 27, in register_into
deephaven-plugins-deephaven-plugins-1  |   File "/opt/deephaven/venv/lib/python3.10/site-packages/deephaven/plugin/utilities/dhe_safe_callback_wrapper.py", line 35, in register
deephaven-plugins-deephaven-plugins-1  |   File "/opt/deephaven/venv/lib/python3.10/site-packages/deephaven/plugin/utilities/dhe_safe_callback_wrapper.py", line 57, in _register_js
deephaven-plugins-deephaven-plugins-1  | 
deephaven-plugins-deephaven-plugins-1  |        at org.jpy.PyLib.callAndReturnValue(Native Method)
deephaven-plugins-deephaven-plugins-1  |        at org.jpy.PyProxyHandler.invoke(PyProxyHandler.java:120)
deephaven-plugins-deephaven-plugins-1  |        at io.deephaven.server.plugin.python.$Proxy16.initialize_all_and_register_into(Unknown Source)
deephaven-plugins-deephaven-plugins-1  |        at io.deephaven.server.plugin.python.PythonPluginsRegistration.registerInto(PythonPluginsRegistration.java:33)
deephaven-plugins-deephaven-plugins-1  |        at io.deephaven.server.plugin.PluginRegistration.registerAll(PluginRegistration.java:41)
deephaven-plugins-deephaven-plugins-1  |        at io.deephaven.server.runner.DeephavenApiServer.run(DeephavenApiServer.java:150)
deephaven-plugins-deephaven-plugins-1  |        at io.deephaven.server.jetty.JettyMain.main(JettyMain.java:24)
```

File: plugins/ui/src/deephaven/ui/__init__.py (hunk 2 above)
Comment (jnumainville) [reply plugins/ui/src/deephaven/ui/__init__.py:12]: I think despite being a rejected standard, many packages just do it anyways... I'll throw it in setup.cfg just to be consistent for now.

File: docker/config/deephaven.prop [hunk 3]
```diff
@@ -3,8 +3,6 @@ includefiles=dh-defaults.prop
 deephaven.console.type=python
 
 # Add all plugins that you want installed here
```
Comment (mofojed) [inline docker/config/deephaven.prop:5]: Either remove this comment
```suggestion
```
Or leave the comment and add an example, e.g.
```suggestion
# Add all JS-only plugins that you want installed here
# For a plugin named `my-plugin`:
#deephaven.jsPlugins.@deephaven/js-plugin-my-plugin=/opt/deephaven/config/plugins/plugins/my-plugin/src/js
```

Given that we're moving away from JS only plugins, I'm inclined to just remove the comment.

**Review by mofojed — CHANGES_REQUESTED** [review]
Minor change, looks good otherwise

**Review by mofojed — APPROVED** [review]

**Pull request merged**